catalyst_voices_packages/catalyst_compression/catalyst_compression_web/assets/js/catalyst_compression.js

```diff
@@ -1,70 +1,86 @@
-const brotli = await import("https://unpkg.com/brotli-wasm@3.0.0/index.web.js?module").then(m => m.default);
-const zstd = await import("https://unpkg.com/@oneidentity/zstd-js@1.0.3/wasm/index.js?module");
+// initialize a web worker for compression works.
+// this is a persistent worker, will last for life of the app.
+const compressionWorker = new Worker(new URL('./catalyst_compression_worker.js', import.meta.url));
 
-// Initializes the zstd module, must be called before it can be used.
-await zstd.ZstdInit();
+const processingIdsPool = new Set();
 
-/// Compresses hex bytes using brotli compression algorithm and returns compressed hex bytes.
-function _brotliCompress(bytesHex) {
-    const bytes = _hexStringToUint8Array(bytesHex);
-    const compressedBytes = brotli.compress(bytes);
-    return _uint8ArrayToHexString(compressedBytes);
-}
+// A simple id generator function. The generated id must be unique across all the processing ids.
+function generateId() {
```

Comment:
How about we just incremented a number instead of generating random ids?

Reply:
Makes sense, that would be far simpler.